core/serialization/i_serializer.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The record type for an object containing functions for
+ *     serializing part of the workspace.
+ */
+
+'use strict';
+
+goog.module('Blockly.serialization.ISerializer');
+goog.module.declareLegacyNamespace();
+
+// eslint-disable-next-line no-unused-vars
+const Workspace = goog.requireType('Blockly.Workspace');
+
+
+/**
+ * Serializes and deserializes a plugin.
+ * @interface
+ */
+class ISerializer {
+  constructor() {
+    /**
+     * A priority value used to determine the order of deserializing plugins.
+     * More positive priorities are deserialized before less positive
+     * priorities. Eg if you have priorities (0, -10, 10, 100) the order of
+     * deserialiation will be (100, 10, 0, -10).
+     * If two plugins have the same priority, they are deserialized in an
+     * arbitrary order relative to each other.
+     * @type {number}
+     */
+    this.priority;
+  }
+
+  /* eslint-disable no-unused-vars, valid-jsdoc */
+
+  /**
+   * Saves the state of the plugin.
+   * @param {!Workspace} workspace The workspace the plugin to serialize is
+   *     associated with.
+   * @return {?} A JS object containing the plugin's state, or null if
+   *     there is no state to record.
+   */
+  save(workspace) {}
```

Comment:
> Hmm it looks like you need them for non-static methods, but not static methods (5.4.9) 

Ah, I didn't realize! I think so far we have been following [this example](https://github.com/google/closure-compiler/wiki/Structural-Interfaces-in-Closure-Compiler#nominal-interfaces)(possibly outdated?). I tried something similar with the class syntax and the compiler was unthrilled. I think it is fine for right now, but if we end up going back to using es5 syntax I would prefer this matched the style of our other interfaces.